pkg/executor/containerexecutor/containerexecutor.go

```diff
@@ -0,0 +1,844 @@
+					}
+					break
+				}
+				c.log.Debug("TailPodLogs stream scan", "out", b, "pod", pod.Name)
+				logs <- b
+			}
+
+			if err != nil {
+				c.log.Errorw("scanner error", "error", err)
+			}
+		}
+	}()
+	return
+}
+
+// GetPodLogError returns last line as error
+func (c *ContainerExecutor) GetPodLogError(ctx context.Context, pod corev1.Pod) (logsBytes []byte, err error) {
+	// error line should be last one
+	return c.GetPodLogs(pod, 1)
+}
+
+// GetLastLogLineError return error if last line is failed
+func (c *ContainerExecutor) GetLastLogLineError(ctx context.Context, pod corev1.Pod) error {
+	l := c.log.With("pod", pod.Name, "namespace", pod.Namespace)
+	log, err := c.GetPodLogError(ctx, pod)
+	if err != nil {
+		return fmt.Errorf("getPodLogs error: %w", err)
+	}
+
+	l.Debugw("log", "got last log bytes", string(log)) // in case distorted log bytes
+	entry, err := output.GetLogEntry(log)
+	if err != nil {
+		return fmt.Errorf("GetLogEntry error: %w", err)
+	}
+
+	c.log.Errorw("got last log entry", "log", entry.String())
+	return fmt.Errorf("error from last log entry: %s", entry.String())
+}
+
+// GetPodLogs returns pod logs bytes
+func (c *ContainerExecutor) GetPodLogs(pod corev1.Pod, logLinesCount ...int64) (logs []byte, err error) {
+	count := int64(100)
+	if len(logLinesCount) > 0 {
+		count = logLinesCount[0]
+	}
+
+	var containers []string
+	for _, container := range pod.Spec.InitContainers {
+		containers = append(containers, container.Name)
+	}
+
+	for _, container := range pod.Spec.Containers {
+		containers = append(containers, container.Name)
+	}
+
+	for _, container := range containers {
+		podLogOptions := corev1.PodLogOptions{
+			Follow:    false,
+			TailLines: &count,
+			Container: container,
+		}
+
+		podLogRequest := c.clientSet.CoreV1().
+			Pods(c.namespace).
+			GetLogs(pod.Name, &podLogOptions)
+
+		stream, err := podLogRequest.Stream(context.TODO())
+		if err != nil {
+			if len(logs) != 0 && strings.Contains(err.Error(), "PodInitializing") {
+				return logs, nil
+			}
+
+			return logs, err
+		}
+
+		defer stream.Close()
+
+		buf := new(bytes.Buffer)
+		_, err = io.Copy(buf, stream)
+		if err != nil {
+			if len(logs) != 0 && strings.Contains(err.Error(), "PodInitializing") {
+				return logs, nil
+			}
+
+			return logs, err
+		}
+
+		logs = append(logs, buf.Bytes()...)
+	}
+
+	return logs, nil
+}
+
+// AbortK8sJob aborts K8S by job name
+func (c *ContainerExecutor) Abort(jobName string) *testkube.ExecutionResult {
+	var zero int64 = 0
+	bg := metav1.DeletePropagationBackground
+	jobs := c.clientSet.BatchV1().Jobs(c.namespace)
+	err := jobs.Delete(context.TODO(), jobName, metav1.DeleteOptions{
+		GracePeriodSeconds: &zero,
+		PropagationPolicy:  &bg,
+	})
+	if err != nil {
+		return &testkube.ExecutionResult{
+			Status: testkube.ExecutionStatusFailed,
+			Output: err.Error(),
+		}
+	}
+	return &testkube.ExecutionResult{
+		Status: testkube.ExecutionStatusCancelled,
+	}
+}
+
+var defaultJobTemplate = `
```

Comment:
We made a significant efforts to remove them to customizable template in helm chart. We should avoid hard coding as much as possible

Reply:
Then why should we hard code it?